### PR TITLE
enable call getDriverModel function in pedestrian entity.

### PR DESCRIPTION
**Pull request by hakuturu583**
Signed-off-by: MasayaKataoka <ms.kataoka@gmail.com>

## Types of PR

- [ ] New Features
- [x] Upgrade of existing features
- [ ] Bugfix

## Description

enable call getDriverModel function in pedestrian entity.

## How to review this PR.

Please check passing all test cases.

### CODE REVIEW

File: simulation/traffic_simulator/include/traffic_simulator/entity/entity_base.hpp [hunk 1]
```diff
@@ -93,9 +93,9 @@ class EntityBase
 
   virtual auto getGoalPoses() -> std::vector<traffic_simulator_msgs::msg::LaneletPose> = 0;
 
-  virtual auto getDriverModel() -> const traffic_simulator_msgs::msg::DriverModel;
+  virtual auto getDriverModel() -> const traffic_simulator_msgs::msg::DriverModel = 0;
```
Comment (yamacir-kit) [inline simulation/traffic_simulator/include/traffic_simulator/entity/entity_base.hpp:96]:
```
virtual auto getDriverModel() -> const traffic_simulator_msgs::msg::DriverModel = 0
```
↓
```
virtual auto getDriverModel() const -> traffic_simulator_msgs::msg::DriverModel = 0
```

File: simulation/traffic_simulator/include/traffic_simulator/entity/misc_object_entity.hpp [hunk 2]
```diff
@@ -81,6 +81,10 @@ class MiscObjectEntity : public EntityBase
     THROW_SEMANTIC_ERROR("requestAcquirePosition function cannot not use in MiscObjectEntity");
   }
 
+  auto getDriverModel() -> const traffic_simulator_msgs::msg::DriverModel override;
```
Comment (yamacir-kit) [inline simulation/traffic_simulator/include/traffic_simulator/entity/misc_object_entity.hpp:84]:
```
auto getDriverModel() -> const traffic_simulator_msgs::msg::DriverModel override;
```
↓
```
auto getDriverModel() const -> traffic_simulator_msgs::msg::DriverModel override;
```

File: simulation/traffic_simulator/include/traffic_simulator/entity/pedestrian_entity.hpp [hunk 3]
```diff
@@ -98,6 +98,10 @@ class PedestrianEntity : public EntityBase
     return parameters.bounding_box;
   }
 
+  auto getDriverModel() -> const traffic_simulator_msgs::msg::DriverModel;
```
Comment (yamacir-kit) [inline simulation/traffic_simulator/include/traffic_simulator/entity/pedestrian_entity.hpp:101]:
```
auto getDriverModel() -> const traffic_simulator_msgs::msg::DriverModel;
```
↓
```
auto getDriverModel() const -> traffic_simulator_msgs::msg::DriverModel;
```

File: simulation/traffic_simulator/src/entity/misc_object_entity.cpp [hunk 4]
```diff
@@ -49,5 +49,15 @@ auto MiscObjectEntity::getCurrentAction() const -> const std::string
   }
   return "";
 }
+
+auto MiscObjectEntity::getDriverModel() -> const traffic_simulator_msgs::msg::DriverModel
```
Comment (yamacir-kit) [inline simulation/traffic_simulator/src/entity/misc_object_entity.cpp:53]:
```
auto MiscObjectEntity::getDriverModel() -> const traffic_simulator_msgs::msg::DriverModel
```
↓
```
auto MiscObjectEntity::getDriverModel() const -> traffic_simulator_msgs::msg::DriverModel
```

File: simulation/traffic_simulator/src/entity/pedestrian_entity.cpp [hunk 5]
```diff
@@ -109,6 +109,16 @@ void PedestrianEntity::setTargetSpeed(double target_speed, bool continuous)
   target_speed_planner_.setTargetSpeed(target_speed, continuous);
 }
 
+auto PedestrianEntity::getDriverModel() -> const traffic_simulator_msgs::msg::DriverModel
```
Comment (yamacir-kit) [inline simulation/traffic_simulator/src/entity/pedestrian_entity.cpp:112]:
```
auto PedestrianEntity::getDriverModel() -> const traffic_simulator_msgs::msg::DriverModel
```
↓
```
auto PedestrianEntity::getDriverModel() const -> traffic_simulator_msgs::msg::DriverModel
```

**Review by yamacir-kit — CHANGES_REQUESTED** [review]

File: simulation/traffic_simulator/include/traffic_simulator/entity/entity_base.hpp (hunk 1 above)
Comment (hakuturu583) [reply simulation/traffic_simulator/include/traffic_simulator/entity/entity_base.hpp]: fixed in https://github.com/tier4/scenario_simulator_v2/pull/650/commits/d3a8c09835cca7d0b623b86317f54f9152eb257e

File: simulation/traffic_simulator/include/traffic_simulator/entity/misc_object_entity.hpp (hunk 2 above)
Comment (hakuturu583) [reply simulation/traffic_simulator/include/traffic_simulator/entity/misc_object_entity.hpp]: fixed in https://github.com/tier4/scenario_simulator_v2/pull/650/commits/d3a8c09835cca7d0b623b86317f54f9152eb257e

File: simulation/traffic_simulator/include/traffic_simulator/entity/pedestrian_entity.hpp (hunk 3 above)
Comment (hakuturu583) [reply simulation/traffic_simulator/include/traffic_simulator/entity/pedestrian_entity.hpp]: fixed in https://github.com/tier4/scenario_simulator_v2/pull/650/commits/d3a8c09835cca7d0b623b86317f54f9152eb257e

File: simulation/traffic_simulator/src/entity/misc_object_entity.cpp (hunk 4 above)
Comment (hakuturu583) [reply simulation/traffic_simulator/src/entity/misc_object_entity.cpp]: fixed in https://github.com/tier4/scenario_simulator_v2/pull/650/commits/d3a8c09835cca7d0b623b86317f54f9152eb257e

File: simulation/traffic_simulator/src/entity/pedestrian_entity.cpp (hunk 5 above)
Comment (hakuturu583) [reply simulation/traffic_simulator/src/entity/pedestrian_entity.cpp]: fixed in https://github.com/tier4/scenario_simulator_v2/pull/650/commits/d3a8c09835cca7d0b623b86317f54f9152eb257e

**Review by yamacir-kit — APPROVED** [review]

**Pull request merged**